src/EnergyPlus/Autosizing/HeatingAirflowUASizing.cc

```diff
@@ -0,0 +1,217 @@
+
+        BaseSizer::initializeWithinEP(state, _compType, _compName, _printWarningFlag, _callingRoutine);
+        this->sizingString = "Heating Coil Airflow For UA"; // TODO: We should make this a constexpr so we don't need this overridden function in each class just for this one line
+    }
+
+
+    void HeatingAirflowUASizer::initializeForSingleDuctZoneTerminal(Real64 const elevation, Real64 mainFlowRate) {
+        this->initializeFromAPI(elevation);
+        this->zoneSizingRunDone = true;
+        this->curZoneEqNum = 1;
+        this->termUnitSingDuct = true;
+        this->curTermUnitSizingNum = 1;
+        this->termUnitSizing.allocate(1);
+        this->termUnitSizing(1).AirVolFlow = mainFlowRate;
+    }
+
+    void HeatingAirflowUASizer::initializeForZoneInductionUnit(Real64 const elevation, Real64 mainFlowRate,
+                                                               Real64 reheatMultiplier) {
+        this->initializeFromAPI(elevation);
+        this->zoneSizingRunDone = true;
+        this->curZoneEqNum = 1;
+        this->termUnitIU = true;
+        this->termUnitPIU = true; // probably don't need to set both, but whatever
+        this->curTermUnitSizingNum = 1;
+        this->termUnitSizing.allocate(1);
+        this->termUnitSizing(1).AirVolFlow = mainFlowRate;
+        this->termUnitSizing(1).ReheatAirFlowMult = reheatMultiplier;
+    }
+
+    void HeatingAirflowUASizer::initializeForZoneFanCoil(Real64 const elevation, Real64 designHeatVolumeFlowRate) {
+        this->initializeFromAPI(elevation);
+        this->zoneSizingRunDone = true;
+        this->zoneEqFanCoil = true;
+        this->curZoneEqNum = 1;
+        this->finalZoneSizing.allocate(1);
+        this->finalZoneSizing(1).DesHeatVolFlow = designHeatVolumeFlowRate;
+    }
+
+    // TODO: What would the zone otherTypeEq include?
+
+    void HeatingAirflowUASizer::initializeForSystemOutdoorAir(Real64 const elevation, Real64 overallSystemMassFlowRate,
+                                                              bool DOAS) {
+        this->initializeFromAPI(elevation);
+        this->curSysNum = 1;
+        this->curOASysNum = 1;
+        if (DOAS) {
+            this->outsideAirSys(1).AirLoopDOASNum = 0;  // the DOAS structure is a zero-based vector, w00t!
+            this->airloopDOAS.emplace_back();
+            this->airloopDOAS[0].SizingMassFlow = overallSystemMassFlowRate;
+        } else {
+            this->finalSysSizing.allocate(1);
+            this->finalSysSizing(1).DesOutAirVolFlow = 0.0; // TODO: what do I do here?
+            this->autoSizedValue = this->finalSysSizing(this->curSysNum).DesOutAirVolFlow;
+        }
+    }
+
+    // TODO: Probably need a reset() call to reinitialize all the flags back to zero/false/etc., call from each initialize method
+
+    void HeatingAirflowUASizer::initializeForSystemMainDuct(Real64 const elevation, Real64 overallSystemVolFlow,
```

Comment:
I'm not sure it makes sense to break all these up or not, so I left them mostly incomplete.  We should discuss.